src/app/shared/services/books.service.ts

```diff
@@ -11,51 +13,55 @@ export class BooksService {
 
   constructor( private firestore: AngularFirestore, private snackBar: MatSnackBar ) { }
 
-  getReadBooks(){
+  getReadBooks(): Observable<DocumentChangeAction<unknown>[]> {
       return this.firestore.collection(Collections.ReadBooks).snapshotChanges();
   }
 
-  addReadBook(data:IReadBook){
+  addReadBook(data: IReadBook): Observable<void> {
       let today = new Date();
       data.date = today.toLocaleDateString();
-      return new Promise<any>((resolve, reject) => {
+
+      return from(
           this.firestore
             .collection(Collections.ReadBooks)
             .add(data)
-            .then(
-                res => { this.showMessage(BookMessages.SuccessAdd) },
-                err => { this.showMessage(BookMessages.ErrorAdd); reject(err) }
-            )
-      });
+      ).pipe(
+          map(res => {
+              this.showMessage(BookMessages.SuccessAdd)
+            }),
+          catchError(error => of(this.showMessage(BookMessages.ErrorAdd)))
+      )
   }
 
-  updateReadBook(id:string, data: Partial<IReadBook>) {
-      return new Promise<any>((resolve, reject) => {
+  updateReadBook(id: string, data: Partial<IReadBook>): Observable<void> {
```

Comment:
We have to return the data returned from the server to keep data actual.
There's a lot of cases when BE makes some post-processing or something.

Reply:
I believe it's resolved now when I use NgRx